intranet/apps/users/models.py

```diff
@@ -1554,6 +1644,65 @@ def clear_cache(self):
         for attr in User.ldap_user_attributes:
             cache.delete(":".join((self.dn, attr)))
             cache.delete(User.create_secure_cache_key(":".join((self.dn, attr))))
+        logger.debug("Clearing cached DB values")
+        if self.cache:
+            UserCache.objects.get(pk=self.cache.pk).delete()
+        self.set_cache()
+
+    def set_cache(self):
+        logger.debug("Setting DB cache using LDAP values")
+        if not self.cache:
+            self.cache = UserCache.objects.create(objectClass=self.user_type, first_name=self.first_name, last_name=self.last_name)
+        self.cache.grade_number = self.grade.number if self.grade else None
+        self.cache.graduation_year = int(self.graduation_year) if self.graduation_year else None
+        bool_gender = None
+        if self.sex:
+            bool_gender = True if self.sex.lower()[:1] == "m" else False
+        self.cache.gender = bool_gender
+        self.cache.save()
+        self.save()
+
+    def get_from_cache(self, attribute):
+        mappings = {
+            "objectClass": "user_type",
+            "gender": "sex",
+            "first_name": "first_name",
+            "last_name": "last_name",
+            "graduation_year": "graduation_year"
+        }
+        try:
+            val = getattr(self.cache, attribute)
+            if not val:
+                raise AttributeError
+            return val
+        except AttributeError:
```

Comment:
don't need the overhead of try-catch, just do something like:
if val: return val
do other stuff